core/src/main/scala/com/avast/clients/storage/compression/ZstdDecompressOutputStream.scala

```diff
@@ -0,0 +1,52 @@
+package com.avast.clients.storage.compression
+
+import com.github.luben.zstd.{ZstdDecompressCtx, ZstdInputStream}
+
+import java.io.OutputStream
+import java.nio.ByteBuffer
+import java.nio.channels.Channels
+
+class ZstdDecompressOutputStream(outputStream: OutputStream) extends OutputStream {
+  private val decompressCtx = new ZstdDecompressCtx()
+  private val outputChannel = Channels.newChannel(outputStream)
+  private val closed = false
+
+  override def write(chunk: Array[Byte]): Unit = {
+    if (closed) {
+      throw new IllegalStateException("Stream is closed")
+    }
+
+    val inputBuffer = ByteBuffer.allocateDirect(chunk.length)
+    val outputBuffer = ByteBuffer.allocateDirect(ZstdInputStream.recommendedDOutSize().toInt)
```

Comment:
Why you don't reuse this buffer? You can just call `clear` every time a new chunk arrives and it won't allocate a new buffer (which BTW you already do at the end of the loop).

Reply:
Good catch 👍 . Will reuse it.